test/metabase/lib/join_test.cljc

```diff
@@ -74,28 +86,36 @@
           venues-category-id-metadata (lib.metadata/field q1 nil "VENUES" "CATEGORY_ID")
           categories-id-metadata      (lib.metadata/stage-column q2 "ID")]
       (testing "lib/join-clause: return a function that can be resolved later"
-        (let [f (lib/join-clause q2 [(lib/= venues-category-id-metadata categories-id-metadata)])]
+        (let [f (lib/join-clause q2 [(lib/= categories-id-metadata venues-category-id-metadata)])]
           (is (fn? f))
           (is (=? {:lib/type    :mbql/join
                    :lib/options {:lib/uuid string?}
                    :stages      [{:lib/type     :mbql.stage/mbql
                                   :source-table (meta/id :venues)}]
                    :conditions  [[:=
                                   {:lib/uuid string?}
-                                  [:field {:lib/uuid string?} (meta/id :venues :category-id)]
-                                  [:field {:base-type :type/BigInteger, :lib/uuid string?} "ID"]]]}
+                                  [:field {:base-type :type/BigInteger, :lib/uuid string?} "ID"]
+                                  [:field {:lib/uuid string?} (meta/id :venues :category-id)]]]}
```

Comment:
Such fuzzy cases only work if the LHS contains the home column and the RHS the joined column, so I had to swap the references.